cli/src/main/java/org/jboss/sbomer/cli/feature/sbom/service/PncService.java

```diff
@@ -128,50 +138,110 @@ public BuildConfiguration getBuildConfig(String buildConfigId) {
+     *         found.
+     */
+    public GroupConfiguration getGroupConfig(String groupConfigId) {
+        log.debug("Fetching GroupConfiguration from PNC with id '{}'", groupConfigId);
+        try {
+            return groupConfigurationClient.getSpecific(groupConfigId);
+        } catch (RemoteResourceNotFoundException ex) {
+            log.warn("GroupConfiguration with id '{}' was not found in PNC", groupConfigId);
+            return null;
+        } catch (RemoteResourceException ex) {
+            throw new ApplicationException(
+                    "GroupConfiguration could not be retrieved because PNC responded with an error",
+                    ex);
+        }
+    }
+
     /**
      * <p>
      * Obtains the {@link ProductVersionRef} for a given PNC {@link Build} identifier.
      * <p>
      *
      * @param buildId The {@link Build} identifier to get the Product Version for.
-     * @return The {@link ProductVersionRef} object for the related or {@code null} in case it is not possible to obtain
-     *         it.
+     * @return A list with {@link ProductVersionRef} objects for the given Build or empty list in case it is not
+     *         possible to obtain any product version.
      */
-    public ProductVersionRef getProductVersion(String buildId) {
+    public List<ProductVersionRef> getProductVersions(String buildId) {
+        List<ProductVersionRef> productVersions = new ArrayList<>();
+
         log.debug("Fetching Product Version information from PNC for build '{}'", buildId);
 
         if (buildId == null) {
-            return null;
+            log.warn("No PNC Build ID provided, interrupting processing");
+            return Collections.emptyList();
         }
+
         Build build = getBuild(buildId);
 
         if (build == null) {
-            log.warn("Build related to the SBOM could not be found in PNC, interrupting processing");
-            return null;
+            log.warn("Build with ID '{}' could not be found in PNC, interrupting processing", buildId);
+            return Collections.emptyList();
         }
 
         log.debug("Build: {}", build);
 
         BuildConfiguration buildConfig = getBuildConfig(build.getBuildConfigRevision().getId());
 
         if (buildConfig == null) {
-            log.warn("BuildConfig related to the SBOM could not be found in PNC, interrupting processing");
-            return null;
+            log.warn("BuildConfig related to Build '{}' could not be found in PNC, interrupting processing", buildId);
+            return Collections.emptyList();
         }
 
         log.debug("BuildConfig: {}", buildConfig);
 
         ProductVersionRef productVersion = buildConfig.getProductVersion();
 
-        if (productVersion == null) {
+        // Product version was found, let's use it!
+        if (productVersion != null) {
+            return List.of(productVersion);
```

Comment:
I am now wondering: should it be exclusive? Meaning, if a buildConfig has a productVersion assigned, should we get anyway all the productVersions linked to any groupConfig, the buildConfig is part of?

Reply:
Like I mentioned, I don't necessarily like the idea of having to support multiple product versions because of the assignments to multiple build configs. The line you highlight is basically preserving old behavior; if we have it explicitly set, then let's use it without asking questions. I was looking at the group config approach as a fallback, but again -- I'm not sure I like it at all. I would probably better make a requirement for SBOMer to work to assign a product version to a BC.

Reply:
Ok , so we can keep the old behavior (use the Product Version assigned to a BC) and fall back to the GroupConfigs, so the PR looks good. I don't actually dislike this approach because there are different ways the teams are using BC and GC, so we are covering more use cases. Thanks!